org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/WorkspaceManager.java

```diff
@@ -60,11 +63,16 @@
 
 	@Inject
 	private IProjectDescriptionFactory projectDescriptionFactory;
+	
+	@Inject 
+	private UriExtensions uriExtensions;
 
 	private BuildManager buildManager;
 
 	private URI baseDir;
```

Comment:
I've the feeling that baseDir should be removed and the list of workspaceFolders should always be used.
If `isSupports..()` is false, the baseDir could be unwrapped from the a single workspaceFolder.
Can we add a protected getter for the workspaceFolders?

Reply:
OK.